compiler/include/ForallStmt.h

```diff
@@ -32,10 +32,14 @@ class ForallStmt : public Stmt
 public:
   bool       zippered()       const; // was 'zip' keyword used?
   AList&     inductionVariables();   // DefExprs, one per iterated expr
-  AList&     iteratedExpressions();  // SymExprs, CallExprs
+  AList&     iteratedExpressions();  // SymExprs, one per iterated expr
   AList&     intentVariables();      // DefExprs of LoopIntentVars
   BlockStmt* loopBody()       const; // the body of the forall loop
 
+  // when originating from a ForLoop
+  bool       iterCallAlreadyTagged() const;  // already has 'tag' actual
+  bool       noFindOuterVars()       const;  // no add'l shadow variables
```

Comment:
It seems odd to me that these two functions both just return fFromForLoop. Anyway, 'noFindOuterVars' seems like a strange method name; maybe it should be "outerVarsHandled" or something... Anyway both of these seem to have more to do with the flow of forall intents / iterator lowering than with ForallStmts fundamentally. For that reason it seems it might be better to just have an accessor for FromForLoop. Or, even better, avoid the need to even store 'fromForLoop' in this AST node at all, if it can become local information for some transformation pass.

Reply:
I added a comment rationalizing the current design.

Replacing fFromForLoop with a HashSet is a good idea. I would like to defer that until later.

Reply:
I renamed the second accessor to needToHandleOuterVars.